src/test/java/seedu/address/logic/commands/RateCommandTest.java

```diff
@@ -0,0 +1,89 @@
+package seedu.address.logic.commands;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_RATING_ARTEMIS;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_RATING_BABYLON;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.address.logic.commands.CommandTestUtil.showBookAtIndex;
+import static seedu.address.testutil.TypicalBooks.getTypicalBookShelf;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_BOOK;
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_BOOK;
+
+import org.junit.Test;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.CommandHistory;
+import seedu.address.logic.UndoStack;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.book.Rating;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for RateCommand.
+ */
+public class RateCommandTest {
+
+    public static final int RATING_STUB = -1;
+
+    private Model model = new ModelManager(getTypicalBookShelf(), new UserPrefs());
+
+    @Test
+    public void execute_invalidBookIndexUnfilteredList_failure() throws Exception {
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredBookList().size() + 1);
+        RateCommand rateCommand = prepareCommand(outOfBoundIndex, VALID_RATING_BABYLON);
+
+        assertCommandFailure(rateCommand, model, Messages.MESSAGE_INVALID_BOOK_DISPLAYED_INDEX);
+    }
+
+    /**
+     * Edit filtered list where index is larger than size of filtered list,
+     * but smaller than size of address book
```

Comment:
Test with index larger than size of filtered list but smaller than size of book shelf